src/dir.targets

```diff
@@ -164,8 +158,39 @@
 
   <!-- On command line run unit tests on CoreCLR after the Test target -->
   <Target Name="RunTestsForProject"
-          AfterTargets="Test"
-          Condition="'$(RunTestsForProject)'=='true'">
+          AfterTargets="CheckTestCategories"
+          Condition="'$(TestDisabled)'!='True'">
+
+    <!-- To enable this scenario we would need to know list of all possible categories -->
+    <Error Condition="'$(RunInnerLoop)'=='True' And '@(RunTestsCategoriesFiltered)'!='InnerLoop'"
+           Text="Running InnerLoop tests with other test categories is currently not supported! To run all tests use: RunTestsWithoutCategories=None" />
+
+    <ItemGroup>
+      <!-- Excluding special category -->
+      <RunTestsWithCategoriesItems Remove="@(RunTestsWithCategoriesItems)" Condition="'%(Identity)'=='InnerLoop'" />
+    </ItemGroup>
+    <PropertyGroup>
+      <XunitOptions Condition="'@(RunTestsWithCategoriesItems)'!=''">$(XunitOptions) -trait category=@(RunTestsWithCategoriesItems, ' -trait category=') </XunitOptions>
+      <XunitOptions Condition="'@(RunTestsWithoutCategoriesItems)'!=''">$(XunitOptions) -notrait category=@(RunTestsWithoutCategoriesItems, ' -notrait category=') </XunitOptions>
+    </PropertyGroup>
+
+    <!-- xUnit command line without coverage enabled -->
+    <PropertyGroup>
```

Comment:
Do these need to be in the target? Can we just add more XUnitOptions at the end of the command line?